nomos-da/network/subnetworks-assignations/src/versions/v1.rs

```diff
@@ -69,6 +69,15 @@ impl MembershipHandler for FillFromNodeList {
     type NetworkId = u16;
     type Id = PeerId;
 
+    fn new_with(&self, members: Vec<PeerId>, addressbook: HashMap<PeerId, Multiaddr>) -> Self {
+        Self {
+            assignations: Self::fill(&members, self.subnetwork_size, self.dispersal_factor),
+            subnetwork_size: self.subnetwork_size,
+            dispersal_factor: self.dispersal_factor,
+            addressbook,
+        }
+    }
```

Comment:
Nit: I think a better naming would be `updated`. Now we just rebuild everything. But implementation wise is wrong. We should just fill up empty spaces from previous subnetwoks.

Reply:
Oh ok, then the implementation is wrong. I assumed we rebuild the whole thing when the update is received. If we need to node the state of the previous to update, then update should be inside the assignation and not behind a sharable arc swap. Let's sync on this pls

Reply:
> Oh ok, then the implementation is wrong and this is very good info. I assumed we rebuild the whole thing when the update is received and this was driving the design decision of this part of solution. If we need to node the state of the previous to update, then update should be inside the assignation and not behind a sharable arc swap. Let's sync on this pls

yes, it is dependant on the assignation algorithm. Which should have the same update actually.

Reply:
I have moved the sync logic and update logic all the way to assignations:
https://github.com/logos-co/nomos/pull/1323/commits/77667bb32b3ce0a43fce88ab4958c92aa174deed.